src/pstl_offload.cpp

```diff
@@ -96,59 +97,311 @@ __original_msize(void* __user_ptr)
     return __orig_msize(__user_ptr);
 }
 
+
+static auto
+__get_original_realloc()
+{
+    using __realloc_func_type = void* (*)(void*, std::size_t);
+
+    static __realloc_func_type __orig_realloc = __realloc_func_type(dlsym(RTLD_NEXT, "realloc"));
+    return __orig_realloc;
+}
+
+inline bool
+__is_ptr_page_aligned(void* __p)
+{
+    return (uintptr_t)__p % __get_memory_page_size() == 0;
```

Comment:
```suggestion
    return (uintptr_t)__p & (__get_memory_page_size() - 1) == 0;
```
Effectively it should be the same, maybe even from code gen perspective if compiler optimizes it well. You can leave the code as is if you like.

Reply:
I'm afraid that no compilers are smart enough to deduce that `__get_memory_page_size()` returns only power of 2 values. So, let's change.